core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -561,6 +591,213 @@ public UpdateNode plan(Update node)
                 outputs);
     }
 
+    public MergeNode plan(Merge merge)
+    {
+        MergeAnalysis mergeAnalysis = analysis.getMergeAnalysis().orElseThrow(() -> new IllegalArgumentException("analysis.getMergeAnalysis() isn't present"));
+
+        List<ColumnHandle> redistributionColumns = mergeAnalysis.getRedistributionColumnHandles();
+        List<List<ColumnHandle>> mergeCaseColumnsHandles = mergeAnalysis.getMergeCaseColumnHandles();
+
+        RelationPlan target = new RelationPlanner(analysis, symbolAllocator, idAllocator, lambdaDeclarationToSymbolMap, metadata, outerContext, session, recursiveSubqueries)
+                .process(merge.getTarget());
+
+        Assignments.Builder projections = Assignments.builder();
+        projections.putIdentities(target.getRoot().getOutputSymbols());
+
+        Symbol presentColumn = symbolAllocator.newSymbol("present", BOOLEAN);
+        projections.put(presentColumn, TRUE_LITERAL);
+
+        target = new RelationPlan(
+                new ProjectNode(idAllocator.getNextId(), target.getRoot(), projections.build()),
+                mergeAnalysis.getTargetTableScope(),
```

Comment:
Do not record this scope in the `mergeAnalysis`, and instead get it directly from the `analysis`.

Reply:
There is no construct to which the scope could be attached.  @martint and I chatted, and he thought this was reasonable.